spec/controllers/rentals_controller_spec.rb

```diff
@@ -86,7 +86,7 @@
           post :create, params: { rental: invalid_create }
         end.to_not change(Rental, :count)
       end
-      it 're-renders the :new template' do
+      it 're-renders the :new template with a generic warning message' do
```

Comment:
Understood that this might be a work in progress, but some additional tests for the new logic would be a 👍